src/tests/intg/getsockopt_wrapper.c

```diff
@@ -45,6 +48,32 @@ static bool peer_is_pam(int fd)
     return NULL != strstr(unix_socket->sun_path, "pipes/pam");
 }
 
+static bool peer_is_sssctl(const struct ucred *cr)
+{
+    char proc_path[32];
+    char cmd_line[255] = { 0 };
+    int proc_fd;
+    int ret;
+
+    if (cr->pid > -1) {
+        snprintf(proc_path, sizeof(proc_path), "/proc/%d/cmdline",
```

Comment:
Hi,

I'd like to suggest to check for errors/truncation here because we might otherwise send an un-terminated string to `open()`.

bye,
Sumit

Reply:
While man page is somewhat vague:
```
The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str.
```
https://en.cppreference.com/w/c/io/fprintf is more specific:
```
The resulting character string will be terminated with a null character, unless bufsz is zero.
```
Simple test confirms that even if truncation occurs, string is still 0-terminated.

But I'll fix code to handle a case when negative value is returned

Reply:
For consistency I'll also change `get_client_cred()` in 'responder_common.c'

Reply:
Done.

Reply:
Thanks